server/api/index.js

```diff
@@ -0,0 +1,64 @@
+const express = require("express");
+const mongoose = require("mongoose");
+const Recipe = require("./models/recipe");
+
+const { ApolloServer, gql } = require("apollo-server-express");
+
+const typeDefs = gql`
+  input RecipeInput {
+    servings: Int,
+    prepTime: String,
+    cookTime: String,
+    ingredients: [IngredientInput],
+    steps: [String],
+    links: [String],
+    name: String
+  }
+  input IngredientInput {
+    measurement: String!,
+    quantity: Int!,
+    name: String!
+  }
+  type Ingredient {
+    measurement: String!,
+    quantity: Int!,
+    name: String!
+  }
+  type Recipe {
+    servings: Int!,
+    prepTime: String,
+    cookTime: String,
+    ingredients: [Ingredient!],
+    steps: [String!],
+    links: [String],
+    name: String!
+  }
+  type Query { recipes: [Recipe] }
+  type Mutation { createRecipe(input: RecipeInput): Recipe }
+`;
+
+const resolvers = {
+    Query: { 
+      recipes: async() => {
+        return await Recipe.find({});
+      } 
+    },
+    Mutation: {
+      createRecipe: async(parent, args) => {
+        return await Recipe.create(args.input);
+      }
+    }
+};
+
+mongoose.connect("mongodb://server:42pickleheads@ds363118.mlab.com:63118/pickleheads-cookbook")
```

Comment:
Let's put this connection string into an environment variable. We shouldn't have DB credentials or any other secret info in version control.

Reply:
Yeah will do